File: evolink/lib/models/match_person.dart
class MatchPerson {
  final String industry;
  final String stack;
  final String userType;
  final String location;
  final String name;
  final String imageUrl;
  final String? portfolioUrl;
  final String? id;
  final String? email;
  final String? description;
  final DateTime? createdAt;

  MatchPerson({
    required this.industry,
    required this.stack,
    required this.userType,
    required this.location,
    required this.name,
    required this.imageUrl,
    this.portfolioUrl,
    this.id,
    this.email,
    this.description,
    this.createdAt,
  });

  // JSON에서 객체 생성
  factory MatchPerson.fromJson(Map<String, dynamic> json) {
    return MatchPerson(
      id: json['id'],
      industry: json['industry'],
      stack: json['stack'],
      userType: json['userType'],
      location: json['location'],
      name: json['name'],
      imageUrl: json['imageUrl'],
      email: json['email'],
      description: json['description'],
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : null,
    );
  }

  // 객체를 JSON으로 변환
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'industry': industry,
      'stack': stack,
      'userType': userType,
      'location': location,
      'name': name,
      'imageUrl': imageUrl,
      'email': email,
      'description': description,
      'createdAt': createdAt?.toIso8601String(),
    };
  }

  // 복사본 생성 (일부 필드만 변경)
  MatchPerson copyWith({
    String? industry,
    String? stack,
    String? userType,
    String? location,
    String? name,
    String? imageUrl,
    String? portfolioUrl,
    String? id,
    String? email,
    String? description,
    DateTime? createdAt,
  }) {
    return MatchPerson(
      industry: industry ?? this.industry,
      stack: stack ?? this.stack,
      userType: userType ?? this.userType,
      location: location ?? this.location,
      name: name ?? this.name,
      imageUrl: imageUrl ?? this.imageUrl,
      portfolioUrl: portfolioUrl ?? this.portfolioUrl,
      id: id ?? this.id,
      email: email ?? this.email,
      description: description ?? this.description,
      createdAt: createdAt ?? this.createdAt,
    );
  }
} 

File: evolink/src/main/java/com/evolink/spring/model/MatchPerson.java
package com.evolink.spring.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "match_persons")
public class MatchPerson {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String industry;

    @Column(nullable = false)
    private String stack;

    @Column(nullable = false)
    private String userType;

    @Column(nullable = false)
    private String location;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String imageUrl;

    private String email;

    private String description;

    private LocalDateTime createdAt = LocalDateTime.now();

    // Getter/Setter
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getIndustry() { return industry; }
    public void setIndustry(String industry) { this.industry = industry; }
    public String getStack() { return stack; }
    public void setStack(String stack) { this.stack = stack; }
    public String getUserType() { return userType; }
    public void setUserType(String userType) { this.userType = userType; }
    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
} 